rpc.go

```diff
@@ -296,10 +299,8 @@ func (t *Topic) listen() {
 			data, err := handler(msg.ReceivedFrom, t.t.String(), msg.Data)
 			if !strings.Contains(t.t.String(), "/_response") {
 				// This is a normal message; respond with data and error
-				go func() {
-					msgID := cid.NewCidV1(cid.Raw, util.Hash(msg.Data))
-					t.publishResponse(msg.ReceivedFrom, msgID, data, err)
-				}()
+				msgID := cid.NewCidV1(cid.Raw, util.Hash(msg.Data))
+				t.publishResponse(msg.ReceivedFrom, msgID, data, err)
```

Comment:
Ended up making `publishResponse` synchronous. It is actually not a blocking call as I anticipated. What it does: 1) call `newTopic`, which returns immediately, and 2) call `topic.Publish`, which blocks when calling `t.p.disc.Bootstrap` https://github.com/libp2p/go-libp2p-pubsub/blob/master/topic.go#L245, but only if `WithReadiness` is set, which we don't. So, synchronous call always close the response topic, and responses are never out of order.

Reply:
Nice, that sounds much better